src/ansys/templates/python/pyansys_advanced/{{cookiecutter.__project_name_slug}}/.github/workflows/ci_cd.yml

```diff
@@ -31,8 +31,16 @@ jobs:
     strategy:
       matrix:
         os: [windows-latest, ubuntu-latest]
-        python-version: ["3.7", "3.8", "3.9", "3.10"]
+        {% set required_minor = cookiecutter.requires_python[2:] | int %}
+        {%- set python_versions = [ ] -%}
+        {%- for minor in range(7, 11) -%}
+        {%- if minor >= required_minor -%}
+        {{ python_versions.append("3." + minor | string ) or ''}}
```

Comment:
It took me a while till I found a way for avoiding printing the list inside the loop. Notice the usage of:

```text
... or '' }}
```

at the end of the instruction.